CommonTools/RecoAlgos/src/PrimaryVertexAssignment.cc

```diff
@@ -37,7 +37,9 @@ PrimaryVertexAssignment::chargedHadronVertex( const reco::VertexCollection& vert
     time = 0.;
     timeReso = -1.;
   }
-  
+  float dzError = track->dzError();
+  if(isnan(dzError)) dzError=100; //a nan in dZerror likely means a badly measured track, assume large dz error 
```

Comment:
please add a LogWarning here

Reply:
there are enough warning already printed out by tracking for that track. 
I think this should be fixed in tracking anyhow and this is just a protection. I will not spend any more time on this PR.

Reply:
what are the warnings? 
Looking at http://cms-unified.web.cern.ch/cms-unified/joblogs/vlimant_ACDC0_Run2017F-v1-Tau-17Nov2017_940_171218_115258_6053/76/DataProcessing/70059d5a-e3e3-11e7-b15a-02163e01c3a1-64-3-logArchive/cmsRun1/cmsRun1-stdout.log
I only see "local error not pos-def", but this is rather non-descriptive and also happens frequently in other cases during normal running.

The reason I ask for a warning (or even a LogError) is to have a chance to actually be able to fix this issue in tracking, especially if we can not do it based on this reported crash in this specific release (which may as well be not the only reason bad values can happen).

Reply:
the error being not pos-def is the issue that creates the nan (sqrt of a negative number), that's the thing to monitor and/or fix.
Why it should be up to the PVAssociator to monitor that the final track collection has positive defined error matrix? If you want such protection I suggest you ask TRK POG to insert a check before they push back the tracks in the generalTrack vector.

Reply:
I can repeat what I said before: ``` "local error not pos-def", but this is rather non-descriptive and also happens frequently in other cases during normal running.```

Reply:
and I repeat too, if it is non descriptive, ask TRK POG to make it more descriptive, if you want a check more specific on "is the track put in the event having not pos-def error matrix" then put such a check in the track producer, not in a random producer reading tracks...

Reply:
fine, but then based on this logic (put a check in the track producer) this PR can not go in.

Reply:
your (O&C) call for this PR not to go in (so you rather keep the crash in production?)
This is a bugfix and I implemented it. The algorithm works perfectly fine and gives the expected results when dzError is not well defined (i.e. nan) so I do not see why this producer should give an Error or Warning.

Reply:
I have suggested a quick fix and I have asked for a LogWarning to indicate that this after all is not the right place to fix the problem.
I agree that the more appropriate place is to fix the problem in the track producer.
It seems like this fix will not solve the miniAOD crash anyways without rerunning the RECO step upstream.